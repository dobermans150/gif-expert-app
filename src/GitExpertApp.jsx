import React from 'react';
import { useState } from 'react/cjs/react.development';
import CategroyAdd from './components/CategroyAdd';
import GifGrid from './components/GifGrid';

const GitExpertApp = () => {
  const [categories, setcategories] = useState(['one punch']);

  /*
  
  
  const handleAdd = () => {
    setcategories([...categories, 'Hunter X Hunter']);
  };
  
  */

  return (
    <>
      <h2>Git Expert App</h2>
      <hr />

      <CategroyAdd setcategories={setcategories} />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GitExpertApp;
