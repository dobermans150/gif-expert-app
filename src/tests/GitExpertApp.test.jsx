import React from 'react';
import { shallow } from 'enzyme';
import GitExpertApp from '../GitExpertApp';

describe('Pruebas GitExpertApp', () => {
  const wrapper = shallow(<GitExpertApp />);

  test('Debe de retornar una pagina ', () => {
    const wrapper = shallow(<GitExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar una lista de categorias', () => {
    const categories = ['One Punch', 'Dragon Ball'];
    const wrapper = shallow(<GitExpertApp />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
