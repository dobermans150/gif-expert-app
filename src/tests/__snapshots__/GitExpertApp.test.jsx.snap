// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GitExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Git Expert App
  </h2>
  <hr />
  <CategroyAdd
    setcategories={[Function]}
  />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas GitExpertApp Debe de retornar una pagina  1`] = `
<Fragment>
  <h2>
    Git Expert App
  </h2>
  <hr />
  <CategroyAdd
    setcategories={[Function]}
  />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
  </ol>
</Fragment>
`;
